### PR TITLE
Fix options.boundaries not being respected (close #1044)

### LINKED ISSUE #1044: Fix options.boundaries not being respected
**Describe the bug**
tackingOptionsParser will always return default boundaries

### PR DESCRIPTION
**Pull request by greg-el**
- #1044

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: plugins/browser-plugin-youtube-tracking/src/helperFunctions.ts [hunk 1]
```diff
@@ -28,40 +28,39 @@ export function trackingOptionsParser(mediaId: string, conf?: MediaTrackingOptio
...
+  defaults.captureEvents = parsedEvents;
 
-    for (let ev of conf.captureEvents) {
-      const youtubeEvent = CaptureEventToYouTubeEvent[ev];
-      if (CaptureEventToYouTubeEvent.hasOwnProperty(ev) && defaults.youtubeEvents.indexOf(youtubeEvent) === -1) {
-        defaults.youtubeEvents.push(youtubeEvent);
-      }
+  for (let ev of defaults.captureEvents) {
+    const youtubeEvent = CaptureEventToYouTubeEvent[ev];
+    if (CaptureEventToYouTubeEvent.hasOwnProperty(ev) && defaults.youtubeEvents.indexOf(youtubeEvent) === -1) {
```
Comment (paulboocock) [inline plugins/browser-plugin-youtube-tracking/src/helperFunctions.ts:51]: Accessing `CaptureEventToYouTubeEvent[ev]` and then checking `CaptureEventToYouTubeEvent.hasOwnProperty(ev)` seems a little odd. It'll be fine but it feels backwards. 

I'd be more tempted to just get rid of the `const` and access `CaptureEventToYouTubeEvent[ev]` twice.

File: plugins/browser-plugin-youtube-tracking/src/helperFunctions.ts [hunk 2]
```diff
@@ -28,40 +28,39 @@ export function trackingOptionsParser(mediaId: string, conf?: MediaTrackingOptio
...
-      } else {
-        parsedEvents.push(ev);
-      }
+  let parsedEvents: EventGroup = [];
+  for (let ev of defaults.captureEvents) {
+    // If an event is an EventGroup, get the events from that group
+    if (EventGroups.hasOwnProperty(ev)) {
+      parsedEvents = parsedEvents.concat(EventGroups[ev]);
+    } else if (!Object.keys(AllEvents).filter((k) => k === ev)) {
+      console.warn(`'${ev}' is not a valid event.`);
```
Comment (paulboocock) [inline plugins/browser-plugin-youtube-tracking/src/helperFunctions.ts:41]: It'd be good to be able to use LOG here. I think the easier way to achieve that is to move this function into index.ts and don't export it.

**Review by paulboocock — APPROVED** [review]
Couple of nit picks but looks good otherwise

**Review by paulboocock — APPROVED** [review]
LGTM assuming the tests pass!

**Pull request merged**